control-plane-operator/controllers/hostedcontrolplane/config/deployment.go

```diff
@@ -93,6 +93,65 @@ func (c *DeploymentConfig) SetColocation(hcp *hyperv1.HostedControlPlane) {
 	}
 }
 
+const (
+	controlPlaneWorkloadTolerationKey = "hypershift.openshift.io/control-plane"
+	controlPlaneNodeLabel             = "hypershift.openshift.io/control-plane"
+)
+
+func clusterKey(hcp *hyperv1.HostedControlPlane) string {
+	return hcp.Namespace
+}
+
+func (c *DeploymentConfig) SetControlPlaneTolerations(hcp *hyperv1.HostedControlPlane) {
+	clusterTolerationKey := fmt.Sprintf("hypershift.openshift.io/%s", clusterKey(hcp))
+	c.Scheduling.Tolerations = []corev1.Toleration{
+		{
+			Key:      controlPlaneWorkloadTolerationKey,
+			Operator: corev1.TolerationOpEqual,
+			Value:    "true",
+			Effect:   corev1.TaintEffectNoSchedule,
+		},
+		{
+			Key:      clusterTolerationKey,
+			Operator: corev1.TolerationOpEqual,
+			Value:    "true",
+			Effect:   corev1.TaintEffectNoSchedule,
+		},
+	}
+
+	clusterNodeLabel := fmt.Sprintf("hypershift.openshift.io/%s", clusterKey(hcp))
+	if c.Scheduling.Affinity == nil {
+		c.Scheduling.Affinity = &corev1.Affinity{}
+	}
+	if c.Scheduling.Affinity.NodeAffinity == nil {
+		c.Scheduling.Affinity.NodeAffinity = &corev1.NodeAffinity{}
+	}
+	c.Scheduling.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = []corev1.PreferredSchedulingTerm{
+		{
+			Weight: 50,
```

Comment:
we could have this magic number in a constant `minWeight` and then the one below can be `minWeight*2`

Reply:
fixed... started with the larger number since 100 is the maximum weight you can use and made the other one maxWeight/2.